src/SDKs/Azure.Base/data-plane/Azure.Base/HashCode.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Azure.Base
+{
+    /// <summary>
+    /// Copied from https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/HashCode.cs
+    /// </summary>
+    public struct HashCodeCombiner
```

Comment:
Could we call it HashCodeBuilder? Combiner is such a strange word.

Reply:
Sounds good